### DISCUSSION TITLE
Is it possible to get videos from my Mux account, rather than only adding the source file to /videos directory?

### DISCUSSION THREAD

**Discussion by henrybabbage** [muxinc/next-video#181] [Q&A] [answered]
If I add a video to Mux via a CMS such as Sanity.io, and then construct the playback URL from Mux and pass it the the `next-video` player src, is that an OK way to do it? I have a use case where I in some parts of my app I will have local videos from the `/videos` dir and in other parts I will have remote videos, in Mux, but uploaded to Mux via the Sanity.io official Mux plugin. So my editors are in Sanity and want to see uploaded videos appear in the frontend. I would be grateful for any advice on whether I am approaching this in a way that makes sense? Thanks so much :)

It seems to work this way:

```
import { cn } from '@/utils/cn'
import Video from 'next-video'
import { VideoAssetDocument } from 'sanity-plugin-mux-input'

type VideoPlayerProps = {
	video: { asset: VideoAssetDocument }
}

const MUX_BASE_URL = 'https://stream.mux.com/'

export default function VideoPlayer({ video }: VideoPlayerProps) {
	const { asset } = video
	const ratio = `!aspect-[${asset?.data?.aspect_ratio}]` ?? '!aspect-video'
	return (
		<Video
			src={`${MUX_BASE_URL}${asset.playbackId}.m3u8`}
			controls={false}
			autoPlay
			muted
			loop
			className={cn(ratio, 'w-full h-full')}
		/>
	)
}
```

Just wondering if there's a better recommended way.

**Comment by luwes** [comment, marked as answer]
hey @henrybabbage, that's a valid question. That's a totally fine way to do it, as long as the string `src` ends in an `.m3u8` extension there will be no attempt to process the video source. 

In this case you'll only be using the default player of next-video, it's almost the same  as using the [lazy loaded React Mux player](https://github.com/muxinc/elements/tree/main/packages/mux-player-react#lazy-loading) 

related PR https://github.com/muxinc/next-video/issues/166

**Reply by henrybabbage** [reply]
Thanks for your feedback @luwes much appreciated!
